### PR TITLE
[Cosmos] Add change feed tests for subpartitioned containers

### PR DESCRIPTION
**Pull request by simorenoh**
This PR contains additional testing to get coverage on using the change feed with sub-partitioning in the SDK. Pretty much just a port of the original `CosmosContainerChangeFeedTest.java` with the necessary changes to use sub-partitioned containers and objects with hierarchical partition keys.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashContainerChangeFeedTest.java [hunk 1]
```diff
@@ -0,0 +1,835 @@
...
+            options.add(CosmosChangeFeedRequestOptions
+                .createForProcessingFromContinuation(continuations.get(i)));
+        }
+
+        drainAndValidateChangeFeedResults(options, null, expectedTotalEventCountAfterUpdates);
+    }
+
+    @Test(groups = { "emulator" }, timeOut = TIMEOUT)
+    public void asyncChangeFeed_fromBeginning_fullFidelity_forFullRange() throws Exception {
+        assertThrows(
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashContainerChangeFeedTest.java:364]: We probably don't need this test, as it doesn't checks anything related to subpartitioning, right?

**Review by kushagraThapar — APPROVED** [review]
LGTM, thanks @simorenoh

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Pull request closed without merging**